protocol/swap/status_manager.go

```diff
@@ -0,0 +1,63 @@
+package swap
+
+import (
+	"sync"
+
+	"github.com/athanorlabs/atomic-swap/common/types"
+)
+
+// statusManager provides lookup for the status channels. Status channels are
+// ephemeral between runs of swapd.
+type statusManager struct {
+	mu             sync.Mutex
+	statusChannels map[types.Hash]chan Status
+}
+
+func newStatusManager() *statusManager {
+	return &statusManager{
+		mu:             sync.Mutex{},
+		statusChannels: make(map[types.Hash]chan Status),
+	}
+}
+
+// getStatusChan returns any existing status channel or a new status channel for
+// reading or writing.
+func (sm *statusManager) getStatusChan(offerID types.Hash) chan Status {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	_, ok := sm.statusChannels[offerID]
+	if !ok {
+		sm.statusChannels[offerID] = newStatusChannel()
+	}
+
+	return sm.statusChannels[offerID]
+}
+
+// GetStatusChan returns any existing status channel or a new status channel for
+// reading only.
+func (sm *statusManager) GetStatusChan(offerID types.Hash) <-chan Status {
+	return sm.getStatusChan(offerID)
+}
+
+// DeleteStatusChan deletes any status channel associated with the offer ID.
+func (sm *statusManager) DeleteStatusChan(offerID types.Hash) {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	delete(sm.statusChannels, offerID)
+}
```

Comment:
No one is using this yet and I'm not entirely clear what the lifecycle should be. For `net_makeOfferAndSubscribe`, I think you want it to live across multiple takers if the first one fails. If we can figure out the lifecycle before merging, that might be good, but I don't think anyone is going to run enough swaps without restarting for this to budge resource usage noticeably.

Reply:
I would expect it to be deleted when the swap ends (no matter the status) and if it's retaken, then a new channel would be created

Reply:
> I would expect it to be deleted when the swap ends (no matter the status) and if it's retaken, then a new channel would be created

It is handled that way now.